File: library/src/main/java/com/jdroid/java/marshaller/DefaultMarshaller.java
package com.jdroid.java.marshaller;

import com.jdroid.java.date.DateUtils;
import com.jdroid.java.utils.StringUtils;

import java.util.Date;
import java.util.Map;

public class DefaultMarshaller implements Marshaller<Object, Object> {
	
	/**
	 * @see com.jdroid.java.marshaller.Marshaller#marshall(java.lang.Object, com.jdroid.java.marshaller.MarshallerMode,
	 *      java.util.Map)
	 */
	@Override
	public Object marshall(Object object, MarshallerMode mode, Map<String, String> extras) {
		Object marshalled = null;
		if (object != null) {
			if (object instanceof Boolean) {
				marshalled = object;
			} else if (object instanceof Date) {
				marshalled = DateUtils.format((Date)object, DateUtils.DEFAULT_DATE_TIME_FORMAT);
			} else if (object instanceof Number) {
				marshalled = object;
			} else {
				marshalled = StringUtils.getNotEmptyString(object.toString());
			}
		}
		return marshalled;
	}
}

File: jdroid-java-core/src/main/java/com/dipien/java/marshaller/DefaultMarshaller.kt
package com.dipien.java.marshaller

import com.dipien.java.date.DateConfiguration
import com.dipien.java.date.DateUtils

import java.util.Date

class DefaultMarshaller : Marshaller<Any, Any> {

    override fun marshall(value: Any?, mode: MarshallerMode?, extras: Map<String, String>?): Any? {
        var marshalled: Any? = null
        if (value != null) {
            if (value is Boolean) {
                marshalled = value
            } else if (value is Date) {
                marshalled = DateUtils.format(value as Date?, DateConfiguration.getDefaultDateTimeFormat())
            } else if (value is Number) {
                marshalled = value
            } else {
                marshalled = com.dipien.java.utils.StringUtils.getNotEmptyString(value.toString())
            }
        }
        return marshalled
    }
}
